crates/matrix-sdk-ui/src/timeline/inner.rs

```diff
@@ -1271,7 +1277,11 @@ fn update_timeline_reaction(
         // Add the remote echo to the related event
         if let Some(event_id) = remote_echo_to_add {
             let own_user_id = own_user_id.to_owned();
-            reaction_group.0.insert((None, Some(event_id.to_owned())), own_user_id);
+            reaction_group.0.insert(
+                EventItemIdentifier::EventId(event_id.clone()),
+                // FIXME: need to get the actual timestamp of that event server side.
+                ReactionSenderData { id: own_user_id, ts: MilliSecondsSinceUnixEpoch::now() },
```

Comment:
@jonnyandrew My understanding is that, at this point, we only have the response from the reaction request, which means that we probably don't have the event synced yet. I suppose we actually can't do better than temporarily using `now()` as the timestamp for the event. 

Can we expect that the event we create here is replaced/corrected when the actual event shows up on sync ? 
If not, my guess is that we should just attach the `event_id` to the local reaction here, and wait for sync to replace with the actual remote event.

Reply:
That is correct that we only have the response from the reaction request here so the event is unlikely to have synced yet.

I'm not 100% sure what will happen when a reaction is synced but from memory, I think a synced reaction will be handled by [`TimelineInner::handle_live_event`](https://github.com/matrix-org/matrix-rust-sdk/pull/2153/files#diff-4638fca028fc6f81c6c740b6c503361149c9c96dfd4589068c9df7fe98a17e3fR378) and finally replace the existing event in [`TimelineEventHandler::handle_reaction`](https://github.com/matrix-org/matrix-rust-sdk/pull/2153/files#diff-4d824f1cf5f42afa6e621978fc1d2cd31f408fbb5a5baa4b080457fbe5dac3b5R417). Maybe we can add some tests to check this behaviour though?

If the sync will replace the timestamp then perhaps its safest to use that as the source of truth?

Reply:
Yeah it looks like we are trying to remove the local echo of the reaction in that block too: https://github.com/matrix-org/matrix-rust-sdk/blob/06600ac53fed058e3920fb9cabf840d1ea73a6d9/crates/matrix-sdk-ui/src/timeline/event_handler.rs#L442

And `insert` will indeed replace remote echo we created here with the one from the sync. Maybe we just need to touch the local echo in case of failure in this code then 🤔, and let the sync replace things when it succeeds.

Reply:
(kinda gets out of the scope of this PR though 😅 )

Reply:
I created an issue to fix that separately https://github.com/matrix-org/matrix-rust-sdk/issues/2175

I'm setting this PR as ready for review, we'll see if this fix seems like it's required before merging this current change (I think it might be fine because the re-evaluated date is probably closer and close enough to reality compared to the date from the previous local echo)